geode-core/src/main/java/org/apache/geode/cache/client/internal/pooling/ConnectionManagerImpl.java

```diff
@@ -301,31 +301,68 @@ public Connection borrowConnection(long acquireTimeout)
     throw new AllConnectionsInUseException();
   }
 
-  /**
-   * Borrow a connection to a specific server. This task currently allows us to break the connection
-   * limit, because it is used by tasks from the background thread that shouldn't be constrained by
-   * the limit. They will only violate the limit by 1 connection, and that connection will be
-   * destroyed when returned to the pool.
-   */
   @Override
-  public PooledConnection borrowConnection(ServerLocation server,
-      boolean onlyUseExistingCnx) throws AllConnectionsInUseException, NoAvailableServersException {
-    PooledConnection connection =
-        availableConnectionManager.useFirst((c) -> c.getServer().equals(server));
-    if (null != connection) {
-      return connection;
-    }
+  public PooledConnection borrowConnection(ServerLocation server, long acquireTimeout,
```

Comment:
As I see this, previous behavior was only to search existing connection in pool, toward designated server. And if we did not find it, then in case onlyUseExistingCnx is set to false we will create new connection, regardless of number of connections.
 My proposal is, simular to borrowConnections(timeout), if we can not find connection in pool, and number of connections did not reach maximum, we create new connection toward designated server. And implement same retry logic from borrowConnections(timeout).

Reply:
@mivanac You have introduced the wait which was not there before, this changes the behavior from caller perspective; It may have impact on the performance of some of the operation.

Reply:
Added new parameter, which as default behavior disables waiting till timeout.